crates/ra_assists/src/fill_struct_fields.rs

```diff
@@ -0,0 +1,120 @@
+use std::fmt::Write;
+
+use hir::{AdtDef, Ty, source_binder};
+use hir::db::HirDatabase;
+
+use ra_syntax::ast::{self, AstNode, Expr};
+
+use crate::{AssistCtx, Assist, AssistId};
+
+pub(crate) fn fill_struct_fields(mut ctx: AssistCtx<impl HirDatabase>) -> Option<Assist> {
+    let struct_lit = ctx.node_at_offset::<ast::StructLit>()?;
+
+    // If we already have existing struct fields, don't provide the assist.
+    // TODO: provide assist for tuple structs
```

Comment:
we can't really do that, lets remove the TODO

Reply:
@matklad Could you elaborate on why we are unable to eventually support this?

Reply:
There's no such thing as a tuple struct literal: `Color(100, 200, 0)` is a function all.